packages/go/cypher/frontend/pattern.go

```diff
@@ -145,13 +148,52 @@ func (s *RelationshipPatternVisitor) ExitOC_RightArrowHead(ctx *parser.OC_RightA
 }
 
 func (s *RelationshipPatternVisitor) EnterOC_RangeLiteral(ctx *parser.OC_RangeLiteralContext) {
-	s.ctx.Enter(&RangeLiteralVisitor{
-		PatternRange: &model.PatternRange{},
-	})
+	const (
```

Comment:
Not the biggest fan of having to write a little state machine for this but I couldn't think of a better way to correctly negotiate pattern ranges.